api/api_ethereum.go

```diff
@@ -1109,18 +1109,12 @@ func (args *EthTransactionArgs) setDefaults(ctx context.Context, b Backend) erro
 				)
 				if isMagma {
 					// After Magma hard fork, `gasFeeCap` was set to `baseFee*2` by default.
-					gasFeeCap = new(big.Int).Mul(gasPrice, big.NewInt(2))
+					gasFeeCap = gasPrice
 				}
 				args.MaxFeePerGas = (*hexutil.Big)(gasFeeCap)
 			}
-			if isMagma {
```

Comment:
Please apply those modifications to `setDefaults` function in `tx_args.go`.

Reply:
I think these cases are different.
In api_ethereum.go, `args.MaxFeePerGas` cannot be a nil because If it is nil, it is assigned some value.
In tx_args.go, if `args.MaxFeePerGas` is not nil, in other words user assigned some value, we need to check args.MaxFeePerGas < baseFee, not gasPrice(baseFee * 2)

`args.MaxFeePerGas < gasPrice` ==> `args.MaxFeePerGas < currentBlockHeader.BaseFee`

Anyway I think we need to refactor these functions...

Reply:
Um.. still i'm confused...
I can't understand why we don't need to check in api_ethereum.go, but have to check in tx_args.go.
I think still possible to set `args.MaxFeePerGas` value by user in api_ethereum.go also.

Reply:
Ohh.. right then we need to remain the `isMagma` condition in `api_ethereum.go`, right?

Reply:
Yes, I also agree with doing the same (or similar) validation check will be better

Reply:
Thansk!